File: lintcode/naive/sort_integers/cpp/sort_integers2.cpp
#include <algorithm>
#include <vector>

class Solution {
public:
    /**
     * @param A an integer array
     * @return void
     */
    void sortIntegers(std::vector<int>& A) 
    {
        // Selection sort.
        
        for (decltype(A.size()) i{ 0 }; i < A.size(); ++i) 
        {
            auto min = i;
            
            for (auto j = i; j < A.size(); ++j) {
                min = std::min(j, min, 
                    [&A] (decltype(j) left, decltype(j) right) {
                        return A[left] < A[right];
                    });
            }
            
            std::swap(A[i], A[min]);
        }
    }
};

File: lintcode/naive/sort_integers/py/sort_integers2.py
class Solution:
    # @param {int[]} A an integer array
    # @return nothing
    def sortIntegers(self, A):
        # Selection sort.
        
        for i in xrange(len(A)):
            j = min(xrange(i, len(A)), key=lambda x: A[x])
            A[i], A[j] = A[j], A[i]
